lib/configurationSchema.js

```diff
@@ -0,0 +1,438 @@
+        stripFileExtensions: { type: 'boolean' },
+        tab: { type: 'boolean' },
+        useRelativePaths: { type: 'boolean' },
+      },
+    },
+    minimumVersion: {
+      default: '0.0.0',
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'string',
+        },
+      ],
+    },
+    moduleNameFormatter: {
+      instanceof: 'Function',
+      default: ({ moduleName }) => moduleName,
+    },
+    moduleSideEffectImports: {
+      instanceof: 'Function',
+      default: () => [],
+    },
+    namedExports: {
+      default: {},
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'object',
+          additionalProperties: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+        },
+      ],
+    },
+    packageDependencies: {
+      default: ({ config }) =>
+        findPackageDependencies(
+          config.workingDirectory,
+          config.get('importDevDependencies')
+        ),
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'array',
+          items: {
+            type: `string`,
+          },
+        },
+      ],
+    },
+    parserPlugins: {
+      default: DEFAULT_PARSER_PLUGINS,
+      type: 'array',
+      items: {
+        anyOf: [
+          {
+            type: 'string',
+          },
+          {
+            type: 'array',
+            minItems: 2,
+            additionalItems: false,
+            items: [
+              {
+                type: 'string',
+              },
+              {
+                type: 'object',
+                additionalProperties: {
+                  type: 'string',
+                },
+              },
+            ],
+          },
+        ],
+      },
+    },
+    sortImports: {
+      default: true,
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'boolean',
+        },
+      ],
+    },
+    stripFileExtensions: {
+      default: ['.js', '.jsx', '.ts', '.tsx'],
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'array',
+          items: {
+            type: 'string',
+          },
+        },
+      ],
+    },
+    tab: {
+      default: '  ',
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'string',
+        },
+      ],
+    },
+    useRelativePaths: {
+      default: true,
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'boolean',
+        },
+      ],
+    },
+  },
+};
+
+function findGlobalsFromEnvironments(environments) {
+  const result = Object.keys(globals.builtin);
+
+  environments.forEach((environment) => {
+    const envGlobals = globals[environment];
+    if (!envGlobals) {
+      return;
+    }
+    result.push(...Object.keys(envGlobals));
```

Comment:
Do we want this to remove duplicates?

Reply:
This function was moved from [configuration.js](https://github.com/Galooshi/import-js/pull/612/files/ae686c28c6cd7d39c63c44368f56caf3f68cdb25#diff-94c307851d4ad36c747a9f883703a4c1f964b82c23320b47899c4e543fd9a8f9L23) because the defaults depend on it.

I would suggest that changes to it be made in another PR, trying to keep this as pure "configuration schema validation" as possible. What do you think?

Reply:
Makes sense